sos/report/plugins/slurm.py

```diff
@@ -0,0 +1,128 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, UbuntuPlugin, RedHatPlugin
+from sos.utilities import is_executable
+
+
+class Slurm(Plugin, UbuntuPlugin, RedHatPlugin):
+
+    short_desc = "Slurm Workload Manager"
+
+    plugin_name = 'slurm'
+    profiles = ('hpc',)
+    packages = (
+        # Ubuntu
+        'slurm-wlm',
+        'slurmd',
+        'slurmdbd',
+        'slurmctld',
+        # EL
+        'slurm',
+        'slurm-slurmctld',
+        'slurm-slurmd',
+        'slurm-slurmdbd',
+    )
+    services = (
+        'slurmd',
+        'slurmdbd',
+        'slurmctld',
+    )
+
+    def setup(self):
+        """ Slurm Workload Manager
+        """
+
+        self.add_copy_spec('/etc/slurm/*.conf')
+
+        if is_executable('sinfo'):
+            self.add_cmd_output([
+                'sinfo --all --list-reasons --long',
+                'sinfo --all --long',
+            ])
+
+        if is_executable('squeue'):
```

Comment:
I think this test might not be needed. If a command isn't executable, `sos` won't store output of it (https://github.com/sosreport/sos/blob/main/sos/utilities.py#L261-L270) - but will add a record to `manifest.json`.

But definitely it is fine to use the test here.